doc/specs/Font features and axes-spec.md

```diff
@@ -0,0 +1,92 @@
+
+1. A font feature tag
+2. A parameter value
+
+A font feature tag is constructed using a 4-character feature tag and the parameter value defines how the feature is applied. For most features, the parameter value is simply treated as a binary value - a value of 0 means the feature is not applied and a non-zero value means the feature is applied. For example, a font feature struct like {'ss03', 1} enables stylistic set 3 for the font and a font feature struct like {'liga', 0} disables ligatures. (Technically, the feature tag is _constructed_ with the 4-character tag and is not the 4-character tag itself, but they are treated the same in the example here for brevity's sake).
+
+Currently, we pass in to DWrite a null value for the list of features to apply to the font. This causes DWrite to automatically apply a ['standard' list](https://github.com/fdwr/TextLayoutSampler/blob/master/DrawableObject.ixx#L802) of font features to the font. Naturally, passing in our own list of font features to DWrite means DWrite will _only_ apply the features we defined, and no longer apply the standard list. Since the standard list contains 11 features, we need to consider how we can allow users to specify 1 additional feature or delete 1 of the standard features without needing to redefine all the others.
+
+We will do this by allowing users to define a dictionary in their settings.json file, where the keys are the 4-character feature tags and the values are the parameter values. This dictionary will then get applied to our internal dictionary (which will contain the standard list of 11 features with their parameter values), meaning that any new key-value pairs will get added to our dictionary and any existing key-value pairs will get updated. Finally, this 'merged' dictionary will be what we use to construct the list of features to pass into DWrite.
+
+### **Axes of variation**
+
+Specifying axes of variation is done in an extremely similar manner to the way font features are specified - a 4-character tag is used to specify which font axis is being modified and a numerical value is provided to specify the value the axis should be set to. For example, {'slnt', 20} specifies that the 'slant' axis should be set to 20.
+
+There is also a standard list of axes of variation, and each axis has its own default. We will approach this the same way we approached font features, by allowing users to specify additional features or omit features without needing to redefine the defaults.
+
+## UI/UX Design
+
+Users will be able to add a new setting to their font objects (added in [#10433](https://github.com/microsoft/terminal/pull/10433)). The resultant font object may look something like this
+
+```
+"font": {
+    "face": "Cascadia Code",
+    "size": 12,

```

Comment:
nit: we should include weight _here_, and not force people to set "axes: wght: 350". However: we need the spec to clarify what happens when there are multiple values and they do not agree.


What is the order in which these are processed:?

`fontWeight` `font.weight` `font.axes[wght]`?

Reply:
Oh dang, this is actually part of the other spec isn't it

Reply:
Between `font.weight` and `fontWeight`, we will [only process `font.weight` and ignore `fontWeight`](https://github.com/microsoft/terminal/pull/10433/files#diff-bbb0e3dd566eb39683a77f3e0fa86afa4a544f72ab7a81ed732f0d48018d6ed3R58-R77). 

Between `font.weight` and `font.axes[wght]`... I'm inclined to say we should ignore `font.weight` and use `font.axes[wght]`, thoughts?

Reply:
I'd agree with @PankajBhojwani here. My reasoning is that if someone went through the trouble of defining `font.axes[wght]`, then _they know what they are doing_. I don't, so I'm probably gonna stick with `fontWeight`, but hey, that's the whole point of layering.

Stick this example in "Potential Issues", and call it answered